sycl/source/detail/queue_impl.hpp

```diff
@@ -255,6 +263,10 @@ class queue_impl {
           make_error_code(errc::invalid),
           "Device provided by native Queue not found in Context.");
     }
+#if XPTI_ENABLE_INSTRUMENTATION
+if (!MHostQueue)
+    xpti::addMetadata(PrepareNotify.traceEvent(), "handle", getHandleRef());
+#endif
```

Comment:
add a proper description to destructor that event and all metadata will be destroyed after